src/workerd/jsg/jsg.h

```diff
@@ -1230,6 +1234,135 @@ void MemoryTracker::trackField(
   trackField(edgeName, value.get(), nodeName);
 }
 
+// A TraceRef is essentially a Ref<T> that is always weak and must be traced during GC.
+template <typename T>
+class TracedRef {
+public:
```

Comment:
This initial iteration was literally just Do Something Quick That demonstrates the concept... and even then it's largely just a copy of `Ref<T>` with the additional debug book keeping.  We need to clarify what the semantics of this alternate-Ref would be and whether it's sufficiently different to warrant a separate class.

FWIW, my goal here is really not to do anything fundamentally different than Ref<T> but to improve our ability to detect when things that should be traced aren't. Something like this is one possible idea.